tests/functional/test_check.py

```diff
@@ -292,3 +292,22 @@ def test_check_include_work_dir_pkg(script):
     )
     assert matches_expected_lines(result.stdout, expected_lines)
     assert result.returncode == 1
+
+
+def test_check_integrity_errors_on_missing_files(data, script, tmpdir):
+    """
+    Work-in-progress failing test for a flag that detects broken packages
+    """
+    to_install = data.packages.joinpath("pip-test-package-0.1.tar.gz")
+    result = script.pip_install_local(to_install)
+    assert 'Successfully installed pip-test-package' in result.stdout
+
+    target = script.site_packages_path / "piptestpackage/__init__.py"
+    target.unlink()
+
+    result = script.pip('check --integrity')
```

Comment:
I would expect this to be a default `check` behavior w/o requiring any extra options. But it's not my call to make so let's ask what pip maintainers think.

cc @pradyunsg @xavfernandez @pfmoore @uranusjr

Reply:
Also, currently, an internal helper assertion fails because it expects a successful outcome. Let's tell it not to expect a return code of 0:
```suggestion
    result = script.pip('check --integrity', expect_error=True)
```

Reply:
About the default behavior, I think that maybe this integrity checking is expensive. While I was frustrated by not being able to write this test, I did some research on how package managers in other languages deal with this kind of checking. Between Elixir's Mix, Nodejs's npm and yarn, only `yarn` does some checking that looks like this and through a flag. Keep in mind this was quick 'research' without much detail, so if we'd like to look for references in another communities, I should re-do it with more care in order to assure that I'm looking in the right place.

As for the second suggestion, I agree!

Reply:
Interesting. Then, it'd be very useful to have at least some of the references that you have posted in the discussion. So that this information is public and would be taken into account when decision is made.